src/views/catalog/CreateFromInstanceTypes/components/SelectInstanceTypeSection/components/CategoryCard/components/InstanceTypesMenu/CustomToggle.tsx

```diff
@@ -0,0 +1,35 @@
+import React, { ElementType, ForwardedRef, forwardRef } from 'react';
+
+export type CustomToggleElement = HTMLDivElement;
+
+export type CustomToggleProps = {
+  className?: string;
+  isExpanded?: boolean;
+  isDisabled?: boolean;
+  innerRef?: ForwardedRef<HTMLDivElement>;
+  onClick?: (...args: any[]) => void;
+  'aria-label'?: string;
+  Component: ElementType;
+};
+
+const CustomToggleBase: React.FC<CustomToggleProps> = ({ ...props }) => {
+  const { Component, isExpanded, isDisabled, innerRef, onClick, 'aria-label': ariaLabel } = props;
+
+  return (
+    <Component
+      type="button"
+      aria-label={ariaLabel}
+      aria-expanded={isExpanded}
+      ref={innerRef}
+      disabled={isDisabled}
+      onClick={onClick}
+      {...props}
+    />
+  );
+};
+
+const CustomToggle = forwardRef<CustomToggleElement, CustomToggleProps>((props, ref) => (
```

Comment:
why do we need all those ref's?

Reply:
`toggleRef` is used along with `menuRef` in the `EnhancedMenuToggle` component to handle key click and mouse click off behavior. It's also used in the `Popper` component in `CustomMenuRenderer` for placement.

`menuRef` is used by the PF `Menu` component.

`containerRef` is passed through `CustomMenuRenderer` to the `Popper` component so that the menu can be appended to the container div created in `CustomMenu`.

For clarity, I adapted this from console code, which I believe is adapted from PF's [composable menu demos](https://www.patternfly.org/v4/demos/composable-menu), and that's how they're designed.